storage-resize-images/functions/src/index.ts

```diff
@@ -49,149 +47,194 @@ logs.init();
+    if (config.deleteOriginalFile === deleteImage.always) {
+      // Delete the original file
+      if (remoteFile) {
+        try {
+          logs.remoteFileDeleting(filePath);
+          await remoteFile.delete();
+          logs.remoteFileDeleted(filePath);
+        } catch (err) {
+          logs.errorDeleting(err);
+        }
+      }
     }
+  }
+};
 
-    if (
-      config.includePathList &&
-      !startsWithArray(config.includePathList, tmpFilePath)
-    ) {
-      logs.imageOutsideOfPaths(config.includePathList, tmpFilePath);
-      return;
-    }
+export const generateResizedImage = functions.storage
+  .object()
+  .onFinalize(async (object) => {
+    await generateResizedImageHandler(object);
+  });
 
-    if (
-      config.excludePathList &&
-      startsWithArray(config.excludePathList, tmpFilePath)
-    ) {
-      logs.imageInsideOfExcludedPaths(config.excludePathList, tmpFilePath);
+/**
+ *
+ */
+export const backfillResizedImages = functions.tasks
+  .taskQueue()
+  .onDispatch(async (data) => {
+    const runtime = getExtensions().runtime();
+    if (!config.doBackfill) {
+      await runtime.setProcessingState(
+        "PROCESSING_COMPLETE",
+        "Existing images were not resized."
+      );
       return;
     }
-
-    if (object.metadata && object.metadata.resizedImage === "true") {
-      logs.imageAlreadyResized();
-      return;
+    if (data == undefined) {
+      logs.startBackfill();
     }
-
-    const bucket = admin.storage().bucket(object.bucket);
-    const filePath = object.name; // File path in the bucket.
-    const parsedPath = path.parse(filePath);
-    const objectMetadata = object;
-
-    let originalFile;
-    let remoteFile;
-    try {
-      originalFile = path.join(os.tmpdir(), filePath);
-      const tempLocalDir = path.dirname(originalFile);
-
-      // Create the temp directory where the storage file will be downloaded.
-      logs.tempDirectoryCreating(tempLocalDir);
-      await mkdirp(tempLocalDir);
-      logs.tempDirectoryCreated(tempLocalDir);
-
-      // Download file from bucket.
-      remoteFile = bucket.file(filePath);
-      logs.imageDownloading(filePath);
-      await remoteFile.download({ destination: originalFile });
-      logs.imageDownloaded(filePath, originalFile);
-
-      // Get a unique list of image types
-      const imageTypes = new Set(config.imageTypes);
-
-      // Convert to a set to remove any duplicate sizes
-      const imageSizes = new Set(config.imageSizes);
-
-      const tasks: Promise<ResizedImageResult>[] = [];
-
-      imageTypes.forEach((format) => {
-        imageSizes.forEach((size) => {
-          tasks.push(
-            modifyImage({
-              bucket,
-              originalFile,
-              parsedPath,
-              contentType,
-              size,
-              objectMetadata: objectMetadata,
-              format,
-            })
-          );
-        });
+    const bucket = admin.storage().bucket(process.env.IMG_BUCKET);
```

Comment:
Can we verify what happens when the bucket entered by the user is mis-spelled etc.?

Reply:
The install fails, which is acceptable behavior IMO: 
<img width="957" alt="Screen Shot 2022-11-29 at 2 02 11 PM" src="https://user-images.githubusercontent.com/4635763/204658120-9b7c77c0-2193-4235-8bfb-4bf6015bd811.png">

Its also worth noting that in console, this is a rich param, and we only let users select real buckets: 
<img width="984" alt="Screen Shot 2022-11-29 at 2 04 08 PM" src="https://user-images.githubusercontent.com/4635763/204658390-92373e97-2466-47ab-b0fb-62fa65a4b7b1.png">